viewing_party/party.py

```diff
@@ -1,23 +1,152 @@
 # ------------- WAVE 1 --------------------
 
+#from tkinter.tix import InputOnly
+
 def create_movie(title, genre, rating):
-    pass
 
-# -----------------------------------------
+    new_movie = {}
+    if title and genre and rating:
+        new_movie["title"] = title
+        new_movie["genre"]= genre
+        new_movie["rating"]= rating
+        return new_movie
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    watched_movies = []
+
+    user_data.append(movie)
+
+    return user_data
+
+def add_to_watched_list(user_data, movie):
+    watchlist = []
+
+    user_data.append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title):
+
+    for items in user_data["watchlist"]:
+        if title in user_data["watchlist"]:
+            user_data["watched"].append(user_data["watchlist"][0])
+            user_data["watchlist"].pop(0)
+        return user_data
+
+# def add_to_watched(user_data, movie):
+#     movies_watched = []
+#     updated_data = []
+
+#     for items in len(user_data)["movie"]:
+#         if movie not in user_data:
+#             user_data["watched"].append(updated_data[movies_watched])
+#         return updated_data
+
+        
+
+
+
+# make a dictionary with these keys above
+#title:title, genre:genre, rating:rating
+
+#-----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
 
+#THIS TEST PASSED AND THEN STOPPED PASSING_ I"M NOT SURE WHY< BUT I DID NOT HAVE TIME TO FIX IT
+def get_watched_avg_rating(user_data):
+    ratings = []
+    ratings_sum = 0
+    if(len(user_data["watched"])) is 0:
+        ratings_average = 0.0
+    else:
+        for i in range(len(user_data["watched"])):
+            ratings.append((user_data["watched"][i]["rating"]))
+            i+=1
+        for rating in ratings:
+            ratings_sum += rating

```

Comment:
While this implementation is still O(n) in time complexity, there's not much benefit in pulling all the ratings out into their own list `ratings` and then going back over that list a second time. (Of course we hadn't covered time complexity when you wrote this, but either way, the second loop is not needed.)

Both can get done in one loop:

```python
for movie in user_data["watched"]:
    ratings_sum += movie["rating"]
```

You would have to change the next line to use `len(user_data["watched"])` but otherwise this would have worked.